pkg/instance/service.go

```diff
@@ -80,6 +80,12 @@ func (s service) SaveInstance(instance *model.DeploymentInstance) error {
 	}
 
 	deployment.Instances = append(deployment.Instances, instance)
+
+	err = s.resolveParameters(deployment)
```

Comment:
I think its best to check `	_, err := s.validateNoCycles(deployment.Instances)` before resolving parameters like we do in `DeployDeployment`. Otherwise you might get a weird behavior in case there are cycles, missing stacks.